scheduler: fix klog.KObjSlice when applied to []*NodeInfo

pohly opened the pull request:
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

The DRA plugin uses klog.KObjSlice on a NodeInfo slice. It didn't actually work and only printed an error message about NodeInfo not implementing klog.KMetata. That's not a compile-time check due to limitations with Go generics and had been missed earlier.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/assign @sanposhiho

pohly committed:
scheduler: fix klog.KObjSlice when applied to []*NodeInfo

The DRA plugin does that. It didn't actually work and only printed an error
message about NodeInfo not implementing klog.KMetata. That's not a compile-time
check due to limitations with Go generics and had been missed earlier.

sanposhiho commented on pkg/scheduler/framework/types.go:
```diff
@@ -591,6 +591,13 @@ type NodeInfo struct {
+// when nodes is a []*NodeInfo.
+var _ klog.KMetadata = &NodeInfo{}
+
+func (n *NodeInfo) GetName() string      { return n.node.Name }
```
should we do `if n != nil && n.node != nil` just in case?

pohly replied:
I don't know whether NodeInfo may have a nil node pointer, but that definitely looks worth checking. Added.

Pushed, please take another look.

pohly committed:
ktesting: support capturing log output

This is a feature of the underlying k8s.io/klog/v2/ktesting which is
useful also when using the Kubernetes ktesting.

sanposhiho commented on test/utils/ktesting/run.go:
Why is it necessary?

pohly replied:
Darn. It isn't. Should have gone into a different PR. Fixed.

sanposhiho replied:
Okay, I thought there was some tricks around go testing that I didn't know of 😅

pohly replied:
I can copy you on the PR where this will come back if you are curious :smile: 

But it's not a priority, so might take a while...

sanposhiho reviewed (commented):
/retest

sanposhiho reviewed (approved):
/lgtm
/approve

sanposhiho commented:
/retest

pacoxu commented:
/test pull-kubernetes-unit 
for https://github.com/kubernetes/kubernetes/issues/125688
